File: backend/schemas/registration.py
from pydantic import BaseModel, EmailStr

class StudentRegistrationRequest(BaseModel):
    name: str
    nick_name: str | None = None
    pronoun: str | None = None
    age: int | None = None
    year: int
    ku_generation: int
    faculty: str
    major: str | None = None
    about_me: str | None = None
    email: EmailStr

class CompanyRegistrationRequest(BaseModel):
    name: str
    website: str | None = None
    logo_url: str | None = None
    location: str | None = None
    description: str | None = None
    contacts: str | None = None

File: frontend/src/types/registration.ts
export interface StudentRegistration {
  name: string;
  nick_name: string;
  pronoun: string;
  age: number | null;
  year: number | null;
  ku_generation: number | null;
  faculty: string;
  major: string;
  about_me: string;
  email: string;
}

export interface CompanyRegistration {
  name: string;
  website: string;
  logo_url: string;
  location: string;
  description: string;
  contacts: string;
}